### PR TITLE
feat: buyer can add product to wishlist(post,get and delete)

### PR DESCRIPTION
**Pull request by emilenfc**
# Description
In this feature, a buyer/ user can add a product to him/her wishlist, (add a product/ get a product and also delete a product from wishlist)

# Type of change

Please check the options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature or Enhancement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have created a loom video for the new feature or enhancement 
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/database/models/Wishes.model.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import { DataTypes } from "sequelize";
+import { sequelize } from ".";
+const Wishlist = sequelize.define(
+  "wishlist",
```
Comment (shyakadavis) [inline src/database/models/Wishes.model.ts:4]:
```suggestion
  "Wishlist",
```

File: src/database/models/Wishes.model.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+      references: {
+        model: "Product",
+        key: "id",
+      },
+      onUpdate: "CASCADE",
+      onDelete: "CASCADE",
+    },
+  },
+  {
+    freezeTableName: false,
```
Comment (shyakadavis) [inline src/database/models/Wishes.model.ts:34]:
```suggestion
    modelName:'Wishlist',
```

**Review by shyakadavis — APPROVED** [review]
LGTM

File: src/controllers/index.ts [hunk 3]
```diff
@@ -1,5 +1,6 @@
...
-export * from './Cart.controller';
-export * from './Product.controller';
-export * from './TwoFactorAuth.controller';
-export * from './User.controller';
+export * from "./Auth.controller";
+export * from "./Cart.controller";
+export * from "./Product.controller";
+export * from "./TwoFactorAuth.controller";
+export * from "./User.controller";
+export * from "./wish.controller";
```
Comment (SergeInGithub) [inline src/controllers/index.ts:6]: Try following the convention we were using of Capital letters on certain file naming
```suggestion
export * from "./Wish.controller";
```

File: src/interfaces/index.ts [hunk 4]
```diff
@@ -1,7 +1,8 @@
...
-export * from './LoggedInUsers.interface';
-export * from './Cart.interface';
+export * from "./MessageResponse";
+export * from "./Payload.interface";
+export * from "./Product.interface";
+export * from "./User.interface";
+export * from "./Token.interface";
+export * from "./LoggedInUsers.interface";
+export * from "./Cart.interface";
+export * from "./wish.interface";
```
Comment (SergeInGithub) [inline src/interfaces/index.ts:8]: Also here
```suggestion
export * from "./Wish.interface";
```

File: src/services/index.ts [hunk 5]
```diff
@@ -1,4 +1,5 @@
-export * from './Auth.service';
-export * from './Cart.service';
-export * from './User.service';
-export * from './Product.service';
+export * from "./Auth.service";
+export * from "./Cart.service";
+export * from "./User.service";
+export * from "./Product.service";
+export * from "./wishes.services";
```
Comment (SergeInGithub) [inline src/services/index.ts:5]:
```suggestion
export * from "./Wishes.services";
```

**Review by musllim — DISMISSED** [review]
that looks good

**Review by SergeInGithub — DISMISSED** [review]
LGTM

**Review by shyakadavis — DISMISSED** [review]
LGTM 🚤 🎆

File: src/database/migrations/20230305183456-create-wishlist-tables.js [hunk 6]
```diff
@@ -0,0 +1,49 @@
+/* eslint-disable no-undef */
```
Comment (mucyoalexaxel) [inline src/database/migrations/20230305183456-create-wishlist-tables.js:1]:
```suggestion
```

File: src/routes/wishes.routes.ts [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+  addWishlistItem,
+  getWishlist,
+  clearWishlist,
+  clearOneWishlistItem,
+} from "../controllers";
+import { isAuth } from "../middleware/auth";
+
+const wishRouter = Router();
+
+wishRouter.post("/:productId", isAuth, addWishlistItem); // add to wishlist
```
Comment (mucyoalexaxel) [inline src/routes/wishes.routes.ts:12]:
```suggestion
wishRouter.post("/:productId", isAuth, addWishlistItem); ```

File: src/routes/wishes.routes.ts [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+
+wishRouter.get("/", [isAuth], getWishlist); // vew all wishes
```
Comment (mucyoalexaxel) [inline src/routes/wishes.routes.ts:14]:
```suggestion
wishRouter.get("/", [isAuth], getWishlist);
```

File: src/services/Wishes.services.ts [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+import { WishListIteminterface } from "../interfaces";
+
+export const addToWishlist = async (wishlistItem: WishListIteminterface) => {
+  const { userId, productId } = wishlistItem;
+
+  const product = await Product.findOne({ where: { id: productId } });
+  if (!product) {
+    throw new Error(`🍎 Product with id ${productId} not found`);
+  }
+  // Check if the product is already in the wishlist
```
Comment (mucyoalexaxel) [inline src/services/Wishes.services.ts:12]:
```suggestion
```

File: src/services/Wishes.services.ts [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+  const isProductInWishlist = await Wishlist.findOne({
+    where: { userId: userId, productId: productId },
+  });
+  if (isProductInWishlist) {
+    throw new Error(
+      `🍎 Product with id ${productId} is already in the wishlist`,
+    );
+  } else {
+    // Add the product to the wishlist
```
Comment (mucyoalexaxel) [inline src/services/Wishes.services.ts:21]:
```suggestion
```

File: src/services/Wishes.services.ts [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+};
+
+export const getAllWishes = async (userId: string) => {
+  const Allwishes = await Wishlist.findAll({
+    where: {
+      userId,
+    },
+  });
+
+  // Get all the productIds in the wishlist
```
Comment (mucyoalexaxel) [inline src/services/Wishes.services.ts:34]:
```suggestion
```

**Review by mucyoalexaxel — CHANGES_REQUESTED** [review]
@emilenfc  remove all the unnecessary comments and the eslint comment under your migration file. 
the rest looks good to me

**Review by mucyoalexaxel — APPROVED** [review]

**Pull request merged**